### ISSUE TITLE
"no viable alternative at input" error for #define with no value

### ISSUE THREAD

**Issue by ebiken** [fattaholmanan/p4-vscode-extension#10]
In P4, there are cases some valiables are defined without value to be used just for decisions (e.g. `#ifndef`).

For example: https://github.com/p4lang/p4c/blob/master/p4include/core.p4#L19

```
#ifndef _CORE_P4_
#define _CORE_P4_
```

will cause error : `no viable alternative at input '_CORE_P4_\n#define'P4 Extension`

Is this intentional? If not, I think it would be better not to show error since this is a valid P4 syntax.

**Comment by fattaholmanan** [comment]
No, it's not intentional! The grammar has to be updated to cover various preprocessors.

**Comment by ebiken** [comment]
Thank you for quick comment. Unfortunately I have no experience in this area so would appreciate if you can fix it when you have time. (Or let me know some pointers I should read and I could study & try to fix it while expecting to take a few months)

**Comment by sadmankiba** [comment]
![p4_directives](https://user-images.githubusercontent.com/56988262/174101626-9c0deafb-ad69-469b-a465-1c0783e3d987.png)

Hi, directives like #if and #endif are not recognized too. Probably will require similar grammar.

**Comment by fattaholmanan** [comment]
This issue has been fixed in the new release.
